packages/blade/src/components/ActionList/ActionListItem.tsx

```diff
@@ -251,6 +251,39 @@ const ActionListItemText = assignWithoutSideEffects(_ActionListItemText, {
   componentId: componentIds.ActionListItemText,
 });
 
+const BaseMenuLeadingItem = ({
+  isSelected,
+  leading,
+  selectionType,
+  isDisabled,
+}: {
+  isSelected?: boolean;
+  leading?: React.ReactNode;
+  selectionType: string | undefined;
+  isDisabled?: boolean;
+}): React.ReactElement | null => {
+  if (selectionType === 'multiple') {
+    return (
+      <BaseBox
+        pointerEvents="none"
+        // Adding aria-hidden because the listbox item in multiselect in itself explains the behaviour so announcing checkbox is unneccesary and just a nice UI tweak for us
+        {...makeAccessible({
+          hidden: true,
+        })}
+      >
+        <Checkbox isChecked={isSelected} tabIndex={-1} isDisabled={isDisabled}>
+          {/*
+        Checkbox requires children. Didn't want to make it optional because its helpful for consumers
+        But for this case in particular, we just want to use Text separately so that we can control spacing and color and keep it consistent with non-multiselect dropdowns
+      */}
+          {null}
+        </Checkbox>
+      </BaseBox>
+    );
+  }
+  return React.isValidElement(leading) ? leading : null;
```

Comment:
This works in scenarios like this too right? - `leading={<span>something</span>}`? or will it break?

Reply:
yes it will 
https://stackblitz.com/edit/sb1-6qdu58nm?file=src%2FApp.tsx